lib/semantics/resolve-names.cc

```diff
@@ -1190,9 +1190,10 @@ void OmpVisitor::ResolveOmpObject(
               // 2.15.3 When a named common block appears in a list, it has the
               // same meaning as if every explicit member of the common block
               // appeared in the list
-              for (Symbol *object :
+              for (const Symbol &object :
                   symbol->get<CommonBlockDetails>().objects()) {
-                ResolveOmp(*object, ompFlag);
+                Symbol &mutableObject{*const_cast<Symbol *>(&object)};
```

Comment:
`CommonBlockDetails::objects_` should really be returning `std::list<common::Reference<Symbol>>`, then we wouldn't have to mess with const-correctness.

Reply:
If we leave it as a const_cast, I think `const_cast<Symbol &>(object)` is better.

Reply:
I've been bitten by casting a non-`const` reference type to a `const &`  causing data copies in ways that I don't fully understand, so I wrote it this way in order feel more confident that copies wouldn't happen.

I'll mess with the API into common blocks; maybe the think to do is to have a distinct interface for mutable symbol references as opposed to constant ones.